### DISCUSSION TITLE
Issue 140 - Read Write permissions for Device Configurations

### DISCUSSION THREAD

**Discussion by gmenziesint** [almenscorner/IntuneCD#141] [General]
Just wanted to provide an update on an issue I raised without re-opening it again.

I've raised a ticket with MS for this issue #140 , it appears that you need `DeviceManagementConfiguration.ReadWrite.All` for https://learn.microsoft.com/en-us/graph/api/intune-deviceconfig-deviceconfiguration-getomasettingplaintextvalue?view=graph-rest-1.0

The docs state that read can be used but as we've noticed this doesn't work correctly.

In the absolute worst case that MS don't provide a fix for this, would it be possible to have a parameter to ignore encrypted OMA settings so that the backup can be run using read only permissions or even an exclusion option to exclude encrypted OMA settings?

Thanks for your help!

**Comment by almenscorner** [comment]
It would definitely be possible to add an argument like `--skip-omasettings`, I would be interested to hear what MS says about it though. I don't get why write permissions would be needed to get these settings :)

**Reply by ztrhgf** [reply]
So please do add such an argument. We don't want to grant write permissions if it is not necessary :)

**Reply by almenscorner** [reply]
It's in the latest beta: https://github.com/almenscorner/IntuneCD/releases/tag/v2.0.5-beta.1

**Reply by gmenziesint** [reply]
Thanks for adding this, still working with MS to understand why this is the case and if they can fix it.

**Reply by gmenziesint** [reply]
I had a case created with MS for this and they've stated it's working as intended and they don't plan to change it, doesn't make any sense but it is what is it, I'll close this now :) Thanks
